File: engines/pageflow-closewindow/app/assets/javascripts/pageflow/closewindow.js
pageflow.widgetTypes.register('closewindow', {
  enhance: function(element) {
    var homeUrl = element.find('a').prop('href');

    // Enable the widget if the visitor came from Home.
    if (document.referrer.indexOf(homeUrl) !== -1) {
      element.addClass('enabled');
    }
  }
})

File: engines/pageflow-closewindow/lib/pageflow/closewindow.rb
require "pageflow/closewindow/engine"
require "pageflow/closewindow/plugin"
require "pageflow/closewindow/widget_type"

module Pageflow
  module Closewindow
    def self.engine
      Engine
    end

    def self.plugin
      Plugin.new
    end

    def self.widget_type
      WidgetType.new
    end
  end
end
